libraries/chain/proposal_evaluator.cpp

```diff
@@ -38,18 +38,27 @@ namespace detail {
 struct proposal_operation_hardfork_visitor
 {
    typedef void result_type;
+   const database& db;
    const fc::time_point_sec block_time;
    const fc::time_point_sec next_maintenance_time;
 
-   proposal_operation_hardfork_visitor( const fc::time_point_sec bt, const fc::time_point_sec nmt )
-   : block_time(bt), next_maintenance_time(nmt) {}
+   proposal_operation_hardfork_visitor(const database& _db, const fc::time_point_sec bt )
+         : db( _db ), block_time(bt), 
+         next_maintenance_time( db.get_dynamic_global_properties().next_maintenance_time ) {}
 
    template<typename T>
    void operator()(const T &v) const {}
 
-   // TODO review and cleanup code below after hard fork
-   // hf_834
    void operator()(const graphene::chain::call_order_update_operation &v) const {
+      if (v.delta_debt.amount > 0 && v.delta_debt.asset_id != asset_id_type( 113 ) // CNY
+            && v.delta_debt.asset_id( db ).bitasset_data_id
+            && !(*( v.delta_debt.asset_id( db ).bitasset_data_id))(db).is_prediction_market )
+      {
+         ilog( "20181206A: proposal contains call_order_update at ${bt}: ${op}", ("bt",block_time)("op",v) );
+         FC_ASSERT( block_time < fc::time_point::now() - fc::seconds(30), "Soft fork - preventing proposal with call_order_update!");
+      }
```

Comment:
I believe the interpretation is: For CNY, prediction markets can have proposals with call_order_update. Otherwise, we ASSERT. I believe the softfork protection should be changed to hardfork protection (backdated to late Dec.?)